jaeger-core/src/main/java/com/uber/jaeger/Span.java

```diff
@@ -267,19 +267,48 @@ private Span setTagAsObject(String key, Object value) {
   }
 
   @Override
+  public Span log(Map<String, ?> fields) {
+    return log(tracer.clock().currentTimeMicros(), fields);
+  }
+
+  @Override
+  public Span log(long timestampMicroseconds, Map<String, ?> fields) {
+    synchronized (this) {
+      if (context.isSampled()) {
+        if (logs == null) {
+          this.logs = new ArrayList<>();
+        }
+        for (Map.Entry<String, ?> kv : fields.entrySet()) {
+          logs.add(new LogData(timestampMicroseconds, kv.getKey(), kv.getValue()));
```

Comment:
also, we don't want to do the conversion of fields here, we just want to stash the map somewhere and only do conversion when the span is reported.